netaddr.go

```diff
@@ -163,58 +163,34 @@ func (err parseIPError) Error() string {
 }
 
 // parseIPv4 parses s as an IPv4 address (in form "192.168.0.1").
-func parseIPv4(in string) (ip IP, err error) {
-	s := in
-	var ip4 [4]byte
-
-	for i := 0; i < 4; i++ {
-		var (
-			j   int
-			acc uint16
-		)
-		// Parse one byte of digits. Bail if we overflow, stop at
-		// first non-digit.
-		//
-		// As of Go 1.15, don't try to factor this digit reading into
-		// a helper function. Its complexity is slightly too high for
-		// inlining, which ends up costing +50% in parse time.
-		for j = 0; j < len(s); j++ {
-			if s[j] < '0' || s[j] > '9' {
-				break
-			}
-			acc = (acc * 10) + uint16(s[j]-'0')
-			if acc > 255 {
-				return IP{}, parseIPError{in: in, msg: "IPv4 field has value >255", at: s}
+func parseIPv4(s string) (ip IP, err error) {
+	var ip4 [4]uint8
```

Comment:
I have also tried to use int64 and shift counter instead of byte array but that did not gave any improvement.